interface/main/tabs/main.php

```diff
@@ -90,15 +90,112 @@ function isEncounterLocked( encounterId ) {
         
         if(isset($_REQUEST['url']))
         {
-
             $tab_one_default=$web_root.$GLOBALS['default_tab_1'];
             if($_REQUEST['url']==='TAB_ONE_DEFAULT')
             {
                 $tab_one_contents=$tab_one_default;
             }
             else
             {
-                $tab_one_contents="../".urldecode($_REQUEST['url']);
+                /*
+                To prevent local file inclusion (LFI) attacks and directory traversals, use a whitelist of acceptable urls that will be compared against.
+                The current list of whitelisted urls (and their required and optional parameters) come from interface/main/main_screen.php.
+                */
+                $inputURL = $_REQUEST['url'];
+                $allowedURLs = array(
+                    "pwd_expires_alert.php",
+                    "../patient_file/summary/demographics.php" => array(
+                        "required" => "set_pid",
+                        "optional" => "set_encounterid"
```

Comment:
I like this approach ...

Reply:
Kinna held up will have a look when I can :)

Reply:
Thank you @muarachmann
Your help is greatly appreciated.

Reply:
Thank you @muarachmann ! I really appreciate this! :)